python/ml4ir/applications/ranking/tests/test_metrics.py

```diff
@@ -192,6 +194,41 @@ def test_acr(self):
         self.assertEquals(ACR()([[1, 0, 0], [0, 0, 1]], [[0.3, 0.6, 0.1], [0.2, 0.2, 0.3]]).numpy(),
                           1.5)
 
+    def test_NDCG(self):
+        # Create an instance of the NDCG metric
+        ndcg_metric = NDCG()
+
+        # Define the true relevance scores and predicted scores
+        y_true = tf.constant([3, 2, 1], dtype=tf.float32)
+        y_pred = tf.constant([0.3, 0.2, 0.1], dtype=tf.float32)
+
+        # Update the metric state
+        ndcg_metric.update_state(y_true, y_pred)
+
+        # Retrieve the result
+        result = ndcg_metric.result().numpy()
+
+        # Expected result: 1.0 (perfect ranking)
+        assert result == 1.0
+
+    def test_NDCG_mask(self):
+        # Create an instance of the NDCG metric
+        ndcg_metric = NDCG()
+
+        # Define the true relevance scores, predicted scores, and mask
+        y_true = tf.constant([3, 2, 1], dtype=tf.float32)
+        y_pred = tf.constant([0.3, 0.5, 0.1], dtype=tf.float32)
+        mask = tf.constant([1, 0, 1], dtype=tf.float32)
+
+        # Update the metric state
+        ndcg_metric.update_state(y_true, y_pred, mask)
+
+        # Retrieve the result
+        result = ndcg_metric.result().numpy()
+
+        # Expected result: 1.0 (only considering the relevant items)
+        assert result == 1.0
+
```

Comment:
@mohazahran Update these tests to look like ranking data batch as opposed to 1 sample.